### DISCUSSION TITLE
client.beginconnet and unity3d

### DISCUSSION THREAD

**Discussion by stevensen4** [inthehand/32feet#362] [General]
Firstly I would like to say that it is great that this software exists.
I used it, but unfortunately I have programming deficits and a problem.

I would like to use the Bluetooth classic functions on a Windows 10 and 11 platform together with Unity3d.
I have a listener and a client and only the client sends data. That works, basically. There is now a problem if the server stops sending, for example, the client does not notice this. For testing, I have implemented the client in a Wpf application, and with the function client.beginconnet together with a callback, I can determine when the connection is lost. Now I have realised that I apparently cannot use Beginconnect in Unity3D.

What other way could I use to determine that the connection to the client has been lost?

**Comment by peterfoot** [comment]
Unfortunately the standard .NET Socket class doesn't work with Bluetooth under the Mono runtime, so the implementation in the 32feet library has to p/invoke the underlying Win32 APIs but it doesn't implement everything in Socket. It may be possible to implement this following the existing open-source .NET code.

**Comment by stevensen4** [comment]
Thank you for your answer.
Unfortunately, I am not able to make such developments.
Is there any other way to determine if the client has lost the connection to the server?

**Reply by peterfoot** [reply]
Are you querying the Available property for the socket - it should throw an exception if the connection becomes broken
